File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/autos/RedLeftSimpleAuto.java
package org.firstinspires.ftc.teamcode.autos;

import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;

import org.firstinspires.ftc.teamcode.HardwareMap;

@Autonomous(name = "[Red Left] Simple Auto (12/32)", group = "Competition")
public class RedLeftSimpleAuto extends LinearOpMode {
    @Override
    public void runOpMode() throws InterruptedException {
        HardwareMap hm = new HardwareMap(this);

        hm.setGrabberIsOpen(false);
        waitForStart();
        hm.resetGrabberEncoder();

        // drive to goal
        hm.driveInches(20);
        hm.turnDegrees(35);

        //deposit on goal
        hm.setGrabberHeight(HardwareMap.GrabberHeight.FIRST_LEVEL);
        hm.waitForGrabber();
        hm.driveInches(7);
        hm.setGrabberIsOpen(true);
        hm.waitForGrabbing();

        //drive to storage unit
        hm.driveInches(-7);
        hm.turnDegrees(-80);
        hm.setGrabberHeight(HardwareMap.GrabberHeight.SECOND_LEVEL);
        hm.waitForGrabber();
        hm.driveInches(19);
        hm.turnDegrees(-40);
        hm.driveInches(8);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/autos/RedLeftSimpleAuto.kt
package org.firstinspires.ftc.teamcode.autos

import com.qualcomm.robotcore.eventloop.opmode.Autonomous
import org.firstinspires.ftc.teamcode.utils.Hardware

@Autonomous(name = "[Red Left] Simple Auto (12/32)", group = "Competition")
class RedLeftSimpleAuto : AutoBase() {

    override fun beforeStart() {
        hw.grabberIsOpen = false
    }

    override fun mainOpMode() {
        hw.resetGrabberEncoder()

        // drive to goal
        hw.driveInches(20.0)
        hw.turnToHeading(35.0)

        //deposit on goal
        hw.grabberHeight = Hardware.GrabberHeight.FIRST_LEVEL
        hw.driveInches(7.0)
        hw.grabberIsOpen = true

        //drive to storage unit
        hw.driveInches(-7.0)
        hw.turnToHeading(-45.0)
        hw.grabberHeight = Hardware.GrabberHeight.SECOND_LEVEL
        hw.driveInches(19.0)
        hw.turnToHeading(-90.0)
        hw.driveInches(8.0)
    }
}